cub/device/dispatch/dispatch_batch_memcpy.cuh

```diff
@@ -215,8 +269,8 @@ __launch_bounds__(int(ChainedPolicyT::ActivePolicy::AgentSmallBufferPolicyT::BLO
                                     BLevBufferOffsetTileState blev_buffer_scan_state,
                                     BLevBlockOffsetTileState blev_block_scan_state)
 {
-  static_assert(std::is_pointer<cub::detail::value_t<InputBufferIt>>::value,
-                "The batched memcpy only supports copying of memory buffers");
+  // static_assert(std::is_pointer<cub::detail::value_t<InputBufferIt>>::value,
+  //               "The batched memcpy only supports copying of memory buffers");
```

Comment:
Since we plan to distinguish `DeviceMemcpy::Batched` and `DeviceCopy::Batched`, I think we can move this assert to the interface method `DeviceMemcpy::Batched`, with the error message hinting at `DeviceCopy::Batched`. E.g.,
```
"DeviceMemcpy::Batched only supports copying of memory buffers. Please consider using DeviceCopy::Batched instead."
```

Reply:
Done.